internal/fs/vss_windows.go

```diff
@@ -0,0 +1,986 @@
+func (p *VssSnapshot) Delete() error {
+	var err error
+	if err = vssFreeSnapshotProperties(&p.snapshotProperties); err != nil {
+		return err
+	}
+
+	for _, mountPoint := range p.mountPointInfo {
+		if mountPoint.isSnapshotted {
+			if err = vssFreeSnapshotProperties(&mountPoint.snapshotProperties); err != nil {
+				return err
+			}
+		}
+	}
+
+	if p.iVssBackupComponents != nil {
+		defer p.iVssBackupComponents.Release()
+
+		err = callAsyncFunctionAndWait(p.iVssBackupComponents.BackupComplete, "BackupComplete",
+			p.timeoutInMillis)
+
+		if _, _, err := p.iVssBackupComponents.DeleteSnapshots(p.snapshotID); err != nil {
+			err = newVssTextError(fmt.Sprintf("Failed to delete snapshot:  %s", err.Error()))
+			p.iVssBackupComponents.AbortBackup()
+		}
+	}
+
+	return err
+}
+
+// asyncCallFunc is the callback type for callAsyncFunctionAndWait.
+type asyncCallFunc func() (*IVSSAsync, error)
+
+// callAsyncFunctionAndWait calls an async functions and waits for it to either
+// finish or timeout.
+func callAsyncFunctionAndWait(function asyncCallFunc, name string, timeoutInMillis uint32) error {
+	iVssAsync, err := function()
+	if err != nil {
+		return err
+	}
+
+	if iVssAsync == nil {
+		return newVssTextError(fmt.Sprintf("%s() returned nil", name))
+	}
+
+	err = iVssAsync.WaitUntilAsyncFinished(timeoutInMillis)
+	iVssAsync.Release()
+	return err
+}
+
+// loadIVssBackupComponentsConstructor finds the constructor of the VSS api
+// inside the VSS dynamic library.
+func loadIVssBackupComponentsConstructor() (*windows.LazyProc, error) {
+	createInstanceName := "?CreateVssBackupComponents@@YAJPEAPEAVIVssBackupComponents@@@Z"
+
+	if runtime.GOARCH == "386" {
+		createInstanceName = "?CreateVssBackupComponents@@YGJPAPAVIVssBackupComponents@@@Z"
+	}
+
+	return findVssProc(createInstanceName)
+}
+
+// findVssProc find a function with the given name inside the VSS api
+// dynamic library.
+func findVssProc(procName string) (*windows.LazyProc, error) {
+	vssDll := windows.NewLazySystemDLL("VssApi.dll")
+	err := vssDll.Load()
+	if err != nil {
+		return &windows.LazyProc{}, err
+	}
+
+	proc := vssDll.NewProc(procName)
+	err = proc.Find()
+	if err != nil {
+		return &windows.LazyProc{}, err
+	}
+
+	return proc, nil
+}
+
+// queryInterface is a wrapper around the windows QueryInterface api.
+func queryInterface(oleIUnknown *ole.IUnknown, guid *ole.GUID) (*interface{}, error) {
+	var ivss *interface{}
+
+	result, _, _ := syscall.Syscall(oleIUnknown.VTable().QueryInterface, 3,
+		uintptr(unsafe.Pointer(oleIUnknown)), uintptr(unsafe.Pointer(guid)),
+		uintptr(unsafe.Pointer(&ivss)))
+	if result != 0 {
+		return nil, newVssError("QueryInterface failed", HRESULT(result))
+	}
+
+	return ivss, nil
+}
+
+// isRunningOn64BitWindows returns true if running on 64-bit windows.
+func isRunningOn64BitWindows() (bool, error) {
+	if runtime.GOARCH == "amd64" {
+		return true, nil
+	}
+
+	handle, err := windows.GetCurrentProcess()
+	if err != nil {
+		return false, err
+	}
```

Comment:
GetCurrentProcess [deprecated](https://godoc.org/golang.org/x/sys/windows#GetCurrentProcess).
Should be `handle := windows.CurrentProcess()`

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/97322db98cdc31a9aba2c358e59809418cdf68be